### DISCUSSION TITLE
Reset-password flow via OTP login in Nextjs

### DISCUSSION THREAD

**Discussion by riccardolardi** [supabase/supabase#29222] [Questions]
How would you go about implementing a reset-password flow via OTP login, where the old password is not required anymore? I have a form at `reset-password/page.tsx`, this should only be accessible if coming from the OTP link sent to the users email, since it doesn't require to enter the old password.

I have another `change-password/page.tsx` route which allows users to change their password, but instead forcing them to verify their identity first using their old password (this is another use case). Now, since the OTP should guarantee the identity, how would it be possible to change the password directly (without needing to enter old password)?

**Comment by silentworks** [comment]
This is in the docs where it explains when using PKCE setup https://supabase.com/docs/guides/auth/passwords?queryGroups=language&language=js&queryGroups=flow&flow=pkce&queryGroups=framework&framework=nextjs#resetting-a-password, you can also check out this working example project which has this flow https://github.com/silentworks/supabase-by-example/tree/main/nextjs
